locked-nft/contracts/NFTLocker.cdc

```diff
@@ -54,17 +60,19 @@ pub contract NFTLocker {
                 self.owner = lockedToken.owner
                 self.lockedAt = lockedToken.lockedAt
                 self.lockedUntil = lockedToken.lockedUntil
-                self.duration = lockedToken.duration
                 self.nftType = lockedToken.nftType
             } else {
                 self.id = id
                 self.owner = owner
                 self.lockedAt = UInt64(getCurrentBlock().timestamp)
                 self.lockedUntil = self.lockedAt + duration
-                self.duration = duration
                 self.nftType = nftType
             }
         }
+
+        pub fun extendLock(extendedDuration: UInt64) {
+            self.lockedUntil = self.lockedUntil + extendedDuration
```

Comment:
could use pre-condition check that it's currently already locked in order to extend

Reply:
Done. Thank you.